request-manager/graph/grapher.go

```diff
@@ -361,26 +410,28 @@ func dependenciesSatisfied(inGraph map[string]bool, dependencies []string) bool
 	return true
 }
 
-// Returns a list of node args that aren't present in the job args map.
-func getMissingArgs(n *spec.Node, jobArgs map[string]bool) ([]string, error) {
+// checkNodeArgs checks whether all node args are present in the job args map.
+func checkNodeArgs(n *spec.Node, jobArgs map[string]bool) error {
 	missing := []string{}
 
-	// Assert that the iterable variable is present.
+	// Assert that the iterable variable is present
 	for _, each := range n.Each {
 		if each == "" {
 			continue
 		}
 		if len(strings.Split(each, ":")) != 2 { // this is malformed input
-			return missing, fmt.Errorf("in node %s: malformed input to `each:`", n.Name)
+			return fmt.Errorf("in node %s: malformed input to `each:`", n.Name)
```

Comment:
A better error message, but let's improve it more. In a software context, avoid "malformed" because it's imprecise and slightly anthropomorphic. For example, another system of ours calls TLS certs "fresh", but that's also imprecise for software. If it was a loaf of bread then sure: fresh or stale. But a TLS cert is valid or expired.